tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1450,26 +1643,38 @@ void CommandQueue::stop_worker() {
 
 void CommandQueue::run_worker() {
     // forever loop checking for commands in the worker queue
+    // Track the worker thread id, for cases where a command calls a sub command.
+    // This is to detect cases where commands may be nested.
+    worker_thread_id = std::hash<std::thread::id>{}(std::this_thread::get_id());
     while (true) {
         if (this->worker_queue.empty()) {
             if (this->worker_state == CommandQueueState::TERMINATE) {
                 break;
             }
-            std::this_thread::yield();
+            std::this_thread::sleep_for(std::chrono::microseconds(10));
```

Comment:
Can you explain this?

Reply:
I profiled some end to end tests to compare `yield` vs `sleep`. Significantly better and consistent perf (up to ~3x) was seen on some unit tests when we used the latter. My theory behind why this happens is because of `yield` relying on the OS to reschedule the thread; which depends on the number of threads currently running, thread priority, etc. Sleeping for a constant period seems to allow the application to schedule its threads and give more deterministic perf.